### PR TITLE
NEXUS-21222: Filter R upload files

### PR DESCRIPTION
**Pull request by chernikovmaksym**
https://issues.sonatype.org/browse/NEXUS-21222

### CODE REVIEW

**Comment by chernikovmaksym** [comment]
Waiting for https://github.com/sonatype-nexus-community/nexus-repository-r/pull/87 to be merged

File: nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RHostedRecipe.groovy [hunk 1]
```diff
@@ -112,11 +110,12 @@ class RHostedRecipe
         .handler(hostedHandlers.putArchive)
         .create())
 
-    builder.route(new Route.Builder()
-        .matcher(BrowseUnsupportedHandler.MATCHER)
-        .handler(browseUnsupportedHandler)
+    builder.route(nonRFileUploadMatcher()
```
Comment (sergeyonofreychuck) [inline nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RHostedRecipe.groovy]: add the security handler

Comment (sergeyonofreychuck) [reply nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RHostedRecipe.groovy]: do we need exception handler?

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RHostedIT.java [hunk 2]
```diff
@@ -85,6 +95,12 @@ public void testFetchPackage() throws Exception
     assertSuccessResponseMatches(resp, AGRICOLAE_PKG_FILE_NAME_131_TGZ);
   }
 
+  @Test
+  public void testFetchNotFoundPackage() throws Exception
```
Comment (sergeyonofreychuck) [inline nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RHostedIT.java]: testFetchNotExistingPackage ?

File: nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/HostedHandlers.java [hunk 3]
```diff
@@ -69,4 +69,9 @@
     context.getRepository().facet(RHostedFacet.class).upload(path, context.getRequest().getPayload());
     return HttpResponses.ok();
   };
+
+  /**
+   * Handle upload non-R file request.
+   */
+  final Handler nonRFileUpload = context -> HttpResponses.badRequest("Non-R file extension or wrong upload path.");
```
Comment (sergeyonofreychuck) [inline nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/HostedHandlers.java]: maybe nonRArchive

File: nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RHostedRecipe.groovy (hunk 1 above)
Comment (chernikovmaksym) [reply nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RHostedRecipe.groovy]: Added security handler in 47bd5fb9dc9524a372cb0a1cd8218dad8aa45f03

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RHostedIT.java (hunk 2 above)
Comment (chernikovmaksym) [reply nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RHostedIT.java]: Done in 47bd5fb9dc9524a372cb0a1cd8218dad8aa45f03

File: nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/HostedHandlers.java (hunk 3 above)
Comment (chernikovmaksym) [reply nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/HostedHandlers.java]: Done in 47bd5fb9dc9524a372cb0a1cd8218dad8aa45f03

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RHostedIT.java [hunk 4]
```diff
@@ -170,7 +186,7 @@ private void uploadPackages(String... names) throws IOException {
     assertThat(getAllComponents(repository), hasSize(names.length));
   }
 
-  private void uploadSinglePackage(String name) throws IOException {
-    client.put(format("%s/%s", PKG_GZ_PATH, name), fileToHttpEntity(name));
+  private HttpResponse uploadSinglePackage(String name) throws IOException {
+    return client.put(format("%s/%s", PKG_GZ_PATH, name), fileToHttpEntity(name));
```
Comment (doddi) [inline nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RHostedIT.java]: Question: does this consume the http response later? for example see https://github.com/sonatype/nexus-internal/blob/128cadc85f603d269f2c0b6d200028117ba00bce/private/testsuite/nexus-repository-testsupport-internal/src/main/java/org/sonatype/nexus/testsuite/testsupport/yum/YumClient.java#L100

Comment (chernikovmaksym) [reply nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RHostedIT.java]: No, updated in 661a917b12dfb9f7554aeccbc23e3b295e57551f

**Pull request merged**